spec/passfort/http_spec.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Passfort::Http do
+  shared_examples "an error response" do
+    before do
+      stub_request(method, api_path).
+        to_return(status: status, body: body)
+    end
+
+    it "raises a specific APIError" do
+      is_expected.to raise_error(instance_of(error_class))
```

Comment:
I would expect us to need to define a `subject` for this to work...

Reply:
The subject is defined in the calling tests, e.g.:
```subject { -> { http.get(path) } }```